lib/handler/redirect-handler.js

```diff
@@ -132,6 +132,14 @@ class RedirectHandler {
     this.opts.maxRedirections = 0
     this.opts.query = null
 
+    // https://tools.ietf.org/html/rfc7231#section-6.4.2
+    // https://fetch.spec.whatwg.org/#http-redirect-fetch
+    // In case of HTTP 301 or 302 with POST, change the method to GET
+    if ((statusCode === 301 || statusCode === 302) && this.opts.method === 'POST') {
+      this.opts.method = 'GET'
+      this.opts.body = null
```

Comment:
If body is a stream it needs to be destroyed.

Reply:
Do we need to do update the HTTP 303 case as well?